### PR TITLE
3.next - Start using pool() internally in Cache

### PR DESCRIPTION
**Pull request by markstory**
Start adopting the new pool() method and PSR16 interface internally. There are a number of current day `Cache` methods that don't fit inside the PSR16 interface. I am considering adding our own interface that encompasses these methods as static tools will complain about
`pool()->increment()` in the future.

The methods that would be needed in this expanded interface are:

- increment
- decrement
- add
- clearGroup

### CODE REVIEW

**Comment by ADmad** [comment]
Tests failing.

File: src/Cache/Cache.php [hunk 1]
```diff
@@ -368,6 +368,7 @@ public static function writeMany($data, $config = 'default')
      */
     public static function read($key, $config = 'default')
     {
+        // In 4.x this needs to change to use pool()
```
Comment (ravage84) [inline src/Cache/Cache.php]: This is actually a "TODO", isn't it?

**Review by ravage84 — COMMENTED** [review]
:+1:

File: src/Cache/Cache.php (hunk 1 above)
Comment (markstory) [reply src/Cache/Cache.php]: It *is* except our lint rules will fail the build on TODOs being present, and I'm frankly too lazy to change them.

Comment (ravage84) [reply src/Cache/Cache.php]: @markstory I thought so ;-)

Would you want to accept TODOs and/or FIXMEs in code during a fast changing development phase like for 3.next and/or 4.x?

Comment (markstory) [reply src/Cache/Cache.php]: I'm fine with it, makes finding future tasks easier to grep for.

Comment (ravage84) [reply src/Cache/Cache.php]: I, personally, love using TODOs in my and our applications' code. But I also understand that you don't want to clutter your code with it.

Comment (markstory) [reply src/Cache/Cache.php]: Yeah there's a balance to be had and I feel we might be too far on one side right now.

Comment (ravage84) [reply src/Cache/Cache.php]: > It is except our lint rules will fail the build on TODOs being present, and I'm frankly too lazy to change them.

I changed that for you with a simple exclusion of this sniff with a comment.

**Review by ravage84 — APPROVED** [review]

**Pull request merged**